samples/subsys/llext/fs_loader/makeimg.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Create a FAT12 disk image to flash in memory
+# This tool requires the following to be available on the host system:
+#
+# - util-linux
+# - dosfstools
+# - mtools
+
+set -e
+
+SECTOR_SIZE=512
+LOGICAL_SECTOR_SIZE=512
+DISK_SIZE_KB=128
+INPUTS=${@:2}
+OUTPUT=$1
+
+
+echo "===> Creating container image"
+fallocate -l ${DISK_SIZE_KB}k "$OUTPUT"
+echo "===> Creating FAT partition image"
+mkfs.fat -F12 -S"$LOGICAL_SECTOR_SIZE" "$OUTPUT" >/dev/null
```

Comment:
How does a user on windows use any of this? Or mac?

Reply:
I'll change 'fallocate' call for 'dd' call.
It works on MSYS2/MSYS and MSYS2/MinGW. Is this what you mean?

Reply:
Well it needs instructions for windows/mac users on how to use this (e.g. readme.rst)

Reply:
There is a _README.rst_ for the _fs_loader_
I'll make this script more user-friendly (+usage & co)

Reply:
Some general shell advice:
- Use shellcheck. shellcheck saves lives.
- Use `set -e` (https://github.com/thesofproject/sof-test/issues/312)
- All code in a function(); use a `main()` function (https://github.com/thesofproject/sof-test/issues/740)

I don't think the Zephyr project has any shell guidelines (for the simple reason that it tries to support vanilla Windows?) but these are small changes that give huge rewards.

Reply:
Hello @marc-hb,
I've written the script to be more friendly (usage & co). It can now runs on GNU OS.

For the 'set-e', I'm not really into that. There are many commands that does not return 0 on success. Moreover, a command that fails in a script does not necessarily means that the script should fail as well. I prefer use 'if'/'else' instead and try to give more information to the user whenever possible (like, why the command failed, maybe why the script fails and maybe give help. Silent failures are not great user experience in my opinion)

Reply:
> For the 'set-e', I'm not really into that. There are many commands that does not return 0 on success.

There are a few but it's very rare actually. Our entire test suite uses `set -e` with great success.

> Moreover, a command that fails in a script does not necessarily means that the script should fail as well.

Yes of course. The fix is very simple: `unusual_command || true`

> I prefer use 'if'/'else' instead and try to give more information to the user whenever possible (like, why the command failed, maybe why the script fails and maybe give help. 

Yes of course. It's not "either ... or... ". `set -e` does absolutely not stop you from doing that. You can and should still do this with `set -e`:

`cmd1 "$arg1" || die "cmd1 %s failed\n" "$arg1"`

`set -e` is for the commands you didn't expect to fail.


> Silent failures are not great user experience in my opinion)

Agreed 100%

Reply:
> How does a user on windows use any of this? Or mac?

The portability of Zephyr's production code and build system is amazing. On the other hand, test code is a much bigger challenge. Last time I checked, twister was quite limited on Windows.

This small shell script is in the `samples/` directory.